packages/block-editor/src/components/list-view/block-select-button.js

```diff
@@ -55,19 +100,80 @@ function ListViewBlockSelectButton(
 	};
 
 	function onKeyDownHandler( event ) {
-		if ( event.keyCode === ENTER || event.keyCode === SPACE ) {
+		// Handle default mode.
+		if (
+			! labelEditingMode &&
+			( event.keyCode === ENTER || event.keyCode === SPACE )
+		) {
 			onClick( event );
 		}
+
+		// Handle Label editing mode.
+		if ( labelEditingMode ) {
+			// Trap events to input when editing to avoid
+			// default list view key handing (e.g. arrow
+			// keys for navigation).
+			event.stopPropagation();
+
+			// Handle ENTER and ESC exits editing mode.
+			if ( event.keyCode === ENTER || event.keyCode === ESCAPE ) {
+				if ( event.keyCode === ENTER ) {
+					// Submit changes only for ENTER.
+					updateBlockAttributes( clientId, {
+						// Include existing metadata (if present) to avoid overwriting existing.
+						metadata: {
+							...( blockAttributes?.metadata &&
+								blockAttributes?.metadata ),
+							name: inputValue,
+						},
+					} );
+				}
+				toggleLabelEditingMode( false );
+			}
+		}
 	}
 
+	useEffect( () => {
```

Comment:
Could you help to understand why this proposal is preferable to the effect?

Reply:
There is also a `__experimentalUseDialog` hook in the compose package that I saw recently and might be of use. That's what's used in popovers and sidebars to manage focus and constrain tabbing.

Reply:
Using autoFocus instead of an effect to auto focus is more semantic it is part of the markup, and devices/browsers could provide an ideal experience, e.g.: a screen reader or braille display knows the field should autofocus and can provide an ideal experience for that. Normally when there is a markup to do some action, I think it is better to use markup that implement the action in javascript.
The other advantage is `const handleFocus = (event) => event.target.select();` will select all the input text each time the field is focused and not just the first time.

Reply:
@jorgefilipecosta thanks for the information. I looked into this and I immediately [received an a11y warning from the linter.](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/bbae2c46ab4ae94122be6c898f2ef313c6154c27/docs/rules/no-autofocus.md)

> Enforce that autoFocus prop is not used on elements. Autofocusing elements can cause usability issues for sighted and non-sighted users, alike.

Reply:
I know there's a linter warning, but what the effect does is exactly the same as what `autoFocus` does.

There are also other hooks like `useFocusOnMount`, which is part of `useDialog` that I mentioned earlier that do the same, so I'm not sure there's a need to implement this explicitly.

My feeling is that the input should be extracted out to a separate component with `useFocusOnMount` behavior implemented there, and `onFocus` triggering `select` as Jorge suggests.

Reply:
`useFocusOnMount` is useful. Thanks for flagging that and apologies for overlooking the `useDialog` mentioned previously.

I've now implemented the pattern and overriden the linting.